src/3d/qgs3dmapscene.cpp

```diff
@@ -893,7 +892,14 @@ void Qgs3DMapScene::setSceneState( Qgs3DMapScene::SceneState state )
 {
   if ( mSceneState == state )
     return;
+
   mSceneState = state;
+  if ( mSceneState == SceneState::Ready )
+  {
+    // compute the correct near/far planes once all the entities are loaded
+    updateCameraNearFarPlanes();
+  }
```

Comment:
I did some more testing and I've realized that the issue only occurs when the 3d scene contains vector layers. To reproduce the issue : 
- create a 3d scene with vector data and display the debug info to see the near/far plane values
- zoom in
- display the configuration dialog and click on ok => the near/far planes are computed
- the new near/far planes are completely off . For example, I get 150000 instead of 2000 for the near plane

You need the first commit from this PR, otherwise the near/far values are not updated in the debug panel

The problem comes from the vector layers. The initial vertical values of the bounding box are very important and therefore the initial guess of the near/far planes are completely off. By adding a `updateCameraNearFarPlanes` in once the scene is loaded, it ensures to get correct near/far plane values because the bounding box coordinates have been updated.

Do you think this issue could be solved with another approach?

Reply:
How about if the near/far planes would be updated when we detect that some bounding boxes have been updated? If that's the reason why near/far plane should be changed, then we should react to that, rather than reacting when the whole scene got loaded?

Reply:
Thanks for the suggestion @wonder-sk . I have completely rewritten this PR based on your suggestion.